customized_list_view/models/custom_list.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2024 ooops404
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import json
+
+from lxml import etree
+
+from odoo import api, fields, models
+
+
+class CustomListView(models.Model):
+    _name = "custom.list.view"
+    _description = "Custom List View"
+
+    name = fields.Char(required=True)
+    model_id = fields.Many2one("ir.model", required=True, ondelete="cascade")
+    model_name = fields.Char(related="model_id.model", store=True)
+    list_view_id = fields.Many2one("ir.ui.view", required=True)
+    line_ids = fields.One2many("custom.list.view.line", "custom_list_view_id")
+    original_arch = fields.Text(readonly=True)
+
+    _sql_constraints = [
+        (
+            "unique_model_list_view_rec",
+            "UNIQUE(model_id, list_view_id)",
+            "Only one record per model is allowed. "
+            "Please modify existing record instead.",
+        )
```

Comment:
This is really "Only one record per view is allowed", and the uniqueness constraint could be applied to `list_view_id` only, IMO.